### PR TITLE
Sippy Analysis

### PR DESCRIPTION
**Pull request by aakankshaduggal**
## Related Issues and Dependencies

…

## This introduces a breaking change

- [ ] Yes
- [x] No

## This Pull Request implements

Recreating Sippy Analysis in a Jupyter Notebook.

## Description

This PR allows to list what analysis output is generated by Sippy to determine if it could be recreated in a notebook environment.

### CODE REVIEW

File: notebooks/sippy-analysis.ipynb [hunk 1]
```diff
@@ -0,0 +1,2411 @@
+{
```
Comment (MichaelClifford) [inline notebooks/sippy-analysis.ipynb:1]: <p>Can you move this notebook to <code>notebooks/data-sources/Sippy/stage/</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aicoe-aiops/ocp-ci-analysis/pull/53/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (MichaelClifford) [inline notebooks/sippy-analysis.ipynb:1]: <p>Is this the best way to get the data? Why aren't we relying on our own data gathering tools from <code>notebooks/data-sources/TestGrid/testgrid_EDA.ipynb</code>  or using our test data at <code>data/raw/</code>?  </p><p> </p><p>Seems to rely too much on Sippy work </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aicoe-aiops/ocp-ci-analysis/pull/53/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (MichaelClifford) [inline notebooks/sippy-analysis.ipynb:1]: <p>A lot of the above is just about getting access to the data which is covered in a notebook, can this be streamlined to only what is needed for you to get the data into your notebook?</p><p> </p><p>I think what is important about this notebook starts below.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aicoe-aiops/ocp-ci-analysis/pull/53/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='25'/>

Comment (MichaelClifford) [inline notebooks/sippy-analysis.ipynb:1]: <p>I would expect the following work would be plots or graphs for each of these items above (Job Pass Rates By Variant, Top Failing Tests Without a Bug,etc). I think that is still missing here.  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aicoe-aiops/ocp-ci-analysis/pull/53/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='26'/>

Comment (MichaelClifford) [inline notebooks/sippy-analysis.ipynb:1]: <p>I'm unclear on this part "...analyzed using the Sippy algorithm" what is the Sippy algorithm?</p><p> </p><p>I see that regex is used to manually categorize many of the tests, but why would that stop us from reproducing that if we have the regex patterns? And I don't think we would need a special regex pattern for all of these, "Job Pass Rates By Job Name" seems to just be the number of times a job passes over the number of times its run grouped by name.</p><p> </p><p>I think we should use the known regex patterns and metrics they've defined to create plots for each of the items you've listed as part of the Sippy work.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aicoe-aiops/ocp-ci-analysis/pull/53/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='35'/>

Comment (aakankshaduggal) [reply notebooks/sippy-analysis.ipynb:1]: <p>Sure, I will fix this. </p>

**Review by MichaelClifford — APPROVED** [review]
LGTM :rocket:

**Pull request merged**